Fix Java Access Bridge compatibility on 32 bit systems

FelipeZanabria opened the pull request:
<!-- Please read and fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/githubPullRequestTemplateExplanationAndExamples.md
Please also note that the NVDA project has a Citizen and Contributor Code of Conduct which can be found at https://github.com/nvaccess/nvda/blob/master/CODE_OF_CONDUCT.md. NV Access expects that all contributors and other community members read and abide by the rules set out in this document while participating or contributing to this project. This includes creating or commenting on issues and pull requests. 

Please initially open PRs as a draft.
When you would like a review, mark the PR as "ready for review". 
See https://github.com/nvaccess/nvda/blob/master/.github/CONTRIBUTING.md.
-->

### Link to issue number:
fixes #10842, fixes #16330
### Summary of the issue:
When NVDA focuses a Java application on a 32-bit system, it gets stuck on the icon, without the ability to interact with it.
### Description of user facing changes
When the user on a 32-bit system focuses on a Java application, the controls can be navigated normally, as was the case in version 2019.2.1 and earlier.
### Description of development approach
- Check the architecture of the running system
- Based on this define the jobj64 class.
     - If the system is 64 bits use the type c_int64, otherwise use c_int.
     - Finally define which dll should be loaded. If the system is 64-bit, load WindowsAccessBridge-32.dll included in NVDA, otherwise load WindowsAccessBridge.dll, available in c:\Windows\System32.
### Testing strategy:
Assuming the user has Java installed on a 32-bit system, simply search for Java in the start menu or control panel. When you find the configure Java element, press enter and move through it
### Known issues with pull request:
None.
### Code Review Checklist:

<!--
This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review of this pull-request.
Check items to confirm you have thought about the relevance of the item.
Where items are missing (eg unit / system tests), please explain in the PR.
To check an item `- [ ]` becomes `- [x]`, note spacing.
You can also check the checkboxes after the PR is created.
A detailed explanation of this checklist is available here:
https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/githubPullRequestTemplateExplanationAndExamples.md#code-review-checklist
-->

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

FelipeZanabria committed:
Fixed Java Access Bridge compatibility on 32 bit systems

FelipeZanabria committed:
Update JABHandler.py to fix Lint test errors

FelipeZanabria committed:
Updated Changes.md to include the Java Access Bridge fix

FelipeZanabria committed:
Fix more Lint issues

seanbudd commented on source/JABHandler.py:
```diff
@@ -43,6 +44,9 @@
 from utils.security import isRunningOnSecureDesktop
 
+#: Verification of the architecture of the running system
+is_64Bit = maxsize > 2**32
```
given #16330, could you make this a system wide constant in `winAPI.constants`?

seanbudd commented on source/JABHandler.py:
```diff
@@ -791,7 +795,7 @@ def event_enterJavaWindow(hwnd):
 	vmID=c_long()
 	accContext=JOBJECT64()
-	timeout=time.time()+0.2
+	timeout = time.time() + 0.5
```
Could you add a comment why this time is used specifically?

FelipeZanabria replied:
This was changed because NVDA would sometimes hang until focus changed.

seanbudd commented on source/JABHandler.py:
```diff
@@ -836,10 +840,15 @@ def enableBridge():
-	global bridgeDll, isRunning
+	global bridgeDll, is_64Bit, isRunning
+	# If the system is 64-bit, load the dll that we have in the NVDA distribution.
+	# Otherwise, it loads the one on the 32-bit system, which does not have the -32 suffix.
```
How can we guarantee this is installed? Shouldn't we be distributing this ourselves?

seanbudd replied:
are you certain we need this other DLL? The one we use is meant to work for 32bit systems. We are specifically using a 32bit DLL.

FelipeZanabria replied:
Yes. First for seeing the code before the change to Python 3, for doing the tests and for having read the jdk code.
He explains that WindowsAccessBridge-32.dll is for accessing 64-bit applications using a 32-bit environment. However, only WindowsAccessBridge.dll is for the 32-bit operating system.

FelipeZanabria replied:
As I explained in the mentioned issue, it would be good to include the dll.
Since it is obtained by compiling JDK code, it is not in my scope. On my local machine I did this, even the code was programmed to look for it in a specific path.
I don't think anyone runs Java 6 or lower.
If we include the dll I think it is better. If Java is updated it will not be blocked by NVDA.

seanbudd replied:
Please follow the instructions in https://github.com/nvaccess/javaAccessBridge32-bin and create a PR updating JAB and including this dll

The Java build we get from azul contains the required `javaaccessbridge.dll`

FelipeZanabria replied:
Now what's left to do is compile the other dll, without the -32 suffix. I have modified the code to point to the same path of the current dll.

seanbudd replied:
you don't need to compile it - you can follow the instructions in the repository to download a compiled version

seanbudd commented on user_docs/en/changes.md:
```diff
@@ -23,7 +23,8 @@
 * NVDA will correctly announce selection changes when editing a cell's text in Microsoft Excel. (#15843)
-* Windows 11 fixes:
+* Java Access Bridge support works as expected on 32-bit systems, as it did in version 2019.2.1 and earlier. (#10842, @FelipeZanabria)
+* * Windows 11 fixes:
```
```suggestion
* Windows 11 fixes:
```

FelipeZanabria committed:
Update user_docs/en/changes.md

Co-authored-by: Sean Budd <seanbudd123@gmail.com>

mwhapples commented:
What environment precisely is this other DLL to be used in? To clarify specifically what I am thinking, is it for a 32-bit OS environment where both NVDA and the Java runtime will be 32-bit, or is it for a 32-bit (java runtime regardless of the OS environment (IE. it is possible to run a 32-bit JRE on a 64-bit OS). Does this correctly detect the actual thing of concern or does it miss a case IE. if the DLL is to be used whenever a 32-bit JRE application is running, then detecting the OS architecture misses the case of 32-bit JRE on 64-bit windows). Can NVDA detect whether the application is a 32-bit or 64-bit process?

FelipeZanabria committed:
Modified the Java handler to include the path to the dll for 32-bit systems in an NVDA lib folder. The way of checking the system's architecture has also been modified for better readability.

FelipeZanabria commented:
> ¿En qué entorno se utilizará precisamente esta otra DLL? Para aclarar específicamente lo que estoy pensando, ¿es para un entorno de sistema operativo de 32 bits donde tanto NVDA como el tiempo de ejecución de Java serán de 32 bits, o es para un entorno de ejecución de Java de 32 bits independientemente del entorno del sistema operativo (es decir, es posible ejecutar un JRE de 32 bits en un sistema operativo de 64 bits). ¿Detecta esto correctamente el problema real o pasa por alto un caso de IE si la DLL se va a utilizar siempre que se ejecuta una aplicación JRE de 32 bits? , luego al detectar la arquitectura del sistema operativo se pasa por alto el caso de JRE de 32 bits en Windows de 64 bits). ¿Puede NVDA detectar si la aplicación es un proceso de 32 o 64 bits?

The first option. The oter option does the current dll.

FelipeZanabria committed:
Update JABHandler.py to fix a comentari line too long

FelipeZanabria commented:
> What environment precisely is this other DLL to be used in? To clarify specifically what I am thinking, is it for a 32-bit OS environment where both NVDA and the Java runtime will be 32-bit?

Yes.

FelipeZanabria commented:
See https://github.com/nvaccess/javaAccessBridge32-bin/pull/2

seanbudd commented on source/JABHandler.py:
```diff
@@ -43,6 +43,9 @@
 from utils.security import isRunningOnSecureDesktop
 
+#: Verification of the architecture of the running system
+is_64Bit = os.environ["PROCESSOR_ARCHITECTURE"].endswith("64")
```
As requested earlier, given https://github.com/nvaccess/nvda/issues/16330, could you make this a system wide constant in winAPI.constants?

seanbudd commented on source/JABHandler.py:
```diff
@@ -791,7 +794,9 @@ def event_enterJavaWindow(hwnd):
 	vmID=c_long()
 	accContext=JOBJECT64()
-	timeout=time.time()+0.2
+	# I changed the timeout to 0.5 because when a Java object disappeared
+	# NVDA would hang until I moved focus again.
```
```suggestion
	# Previous value of 0.2 was increased to 0.5 as when a Java object disappeared
	# NVDA would hang until focus was moved again #16557.

```

seanbudd committed:
Update source/JABHandler.py

josephsl commented:
Hi, if you want to know system architecture, do NOT use “PROCESSOR_ARCHITECTURE” environment variable as it will , if yoㅕsystem architecture, 애 NOT use PROCESSOR_ARCHITECTURE environment variable as it returns whatever the architecture is set for the current process only. In addition to making this an API constant, a much better method is winVersion.getWinVer().processorArchitecture.endswith(“64”), and this call will be useful for a while as we move toward 64-bit NVDA (see #16304). Thanks.

seanbudd commented:
Thanks @FelipeZanabria, can you please pull in the latest changes from https://github.com/nvaccess/javaAccessBridge32-bin/pull/2 and also update https://github.com/nvaccess/nvda/blob/e7e8233fc8fe6a8a1f02002cbe9a142ca2f95ad2/projectDocs/dev/createDevEnvironment.md#L86

seanbudd commented:
Please also add a message to changes.md like this one https://github.com/nvaccess/nvda/blob/57ce236a8b28de5ce788d5582040238396c075d3/user_docs/en/changes.md#L205

FelipeZanabria commented:
Regarding PR #2 from the other repository, I don't know what to do. The call to the last dll is already implemented in JABHandler.py.
Regarding the constant in winApi.constants, Can I write the code that Josefsl provided as a variable outside of the 2 classes? Can I update everything in this PR?

FelipeZanabria committed:
Update createDevEnvironment.md to mentione JDK 17.0.11+9Zulu (17.50.19)

FelipeZanabria committed:
Update changes.md to mentione Java Access Bridge update (17.0.11+9Zulu (17.50.19).)

seanbudd commented:
>Regarding PR 2 from the other repository, I don't know what to do. The call to the last dll is already implemented in JABHandler.py.

You need to include it in this PR, you can do this by the following

```sh
cd include/javaAccessBridge32
git pull
cd ../..
git add include/javaAccessBridge32
git commit -m "update java access bridge dependency"
```

>Regarding the constant in winApi.constants, Can I write the code that Josefsl provided as a variable outside of the 2 classes? Can I update everything in this PR?

Yep, just add the line to that file

FelipeZanabria commented:
I don't have git.

seanbudd commented:
I'm not sure how to help here then.
Please consider following the [contributor process](https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/contributing.md#overview-of-contribution-process) before opening pull requests.

gerald-hartig commented:
Hi @FelipeZanabria. The good news is that git is free, easy to install and there is lots of information available on how to use it. It's an essential part of developing many, many different software, including NVDA. Our documentation includes instructions on how to use git with NVDA. https://github.com/nvaccess/nvda/blob/master/projectDocs/dev/createDevEnvironment.md

FelipeZanabria committed:
Create manual.yml to run submodule update

FelipeZanabria committed:
Delete .github/workflows directory

FelipeZanabria commented:
I didn't download Git because I also run w7, and because my hdd is failing, so it's very slow.
I thought that the web interface would be enough for me. In fact I can do everything except update the required submodule. I can even continue with the constant in winApi and hopefully we close this.
I don't know if it's possible for someone with Git to update the submodule. I've realized that it can be a bit disorganized when creating 2 PRs, and at the same time uncomfortable for reviewers/developers.

seanbudd commented:
@FelipeZanabria - I can cover it this time, but can I ask how you tested this without a dev environment?

seanbudd committed:
update binaries

seanbudd commented:
When the PR build completes you can find the installer artifact from here: https://ci.appveyor.com/project/NVAccess/nvda/builds/49878888

You should be able to test with this build to confirm it works.

seanbudd commented:
https://ci.appveyor.com/api/buildjobs/xw9r4f83r8dacj89/artifacts/output/nvda_snapshot_pr16557-32127,a894d102.exe

FelipeZanabria commented:
I downloaded the latest JABHandler.py from the source code, and checked that it was the same as version 2023.3.4. Since it was the same, I compared it with one from before 2019.3, and I realized that it was missing a reference to the last incorporated dll.
So I edited the source file with what was missing and tried like this:
I removed JABHandler.pyc from librari.zip in the NVDA root, put the source file in the root, because the compiled file is also in the root of the zip.
Restarting NVDA loaded the source file with the changes.
To delete the source file and keep the compiled one, I went into the __pycache folder that was created in the root of NVDA, removed the suffix added by python when generating the bytecode, and put it back into librari.zip.
For the winAPI package, I downloaded it completely, created the folder referring to the package in the root of NVDA, removed it from the zip, modified constants.py, restarted NVDA and with the python console I imported the entire package to create the bytecode, to re-include in the .zip.
If NVDA failed due to a coding error, I started a portable that I had on hand, looked at the old log and continued coding, even though the changes were minimal.
For the lintTest, I have noticed the errors in the compilation of Appveyor.

FelipeZanabria committed:
Added a constant to check the architecture of running system.

Closes #16330

josephsl commented:
Hi,

I see. While I'm not from NV Access, I would like to strongly recommend doing the following before contributing future pull requests:

1. UPGRADE TO (or BUY) Windows 10, preferably Version 22H2 (2022 Update/build 19045); or even better, if you are able, get a system with Windows 11 Version 23H2 (2023 Update) installed so you can be on the same page as most developers (NV Access folks included).
2. INSTALL Git LOCALLY! As you have learned, there are tasks requiring local Git on your computer (one important task is updating submodules).
3. If able, try setting up the local development environment as close to what NV Access folks and contributors are using (Python 3.11 32-bit, Visual Studio (or VS build tools) 2022, to name a few). You can use Git for Windows, or if you would like to do so, try using Windows Subsystem for Linux (WSL) with a distro such as Ubuntu with Git and other tools installed. You can also try using Visual studio Code for writing NVDA contributions.

Thanks.

FelipeZanabria committed:
Update JABHandler.py to use winAPI.constants.is_64Bit

FelipeZanabria committed:
Update changes.md to mentione the `is_64Bit` in `winAPI.constants`change

FelipeZanabria commented:
I can't test, but I checked the contents of the installer with 7-zip, and I see that the windowsAccessBridge.dll file is not found in lib/version.

FelipeZanabria committed:
Update sconstruct to include windowsAccessBridge.dll

seanbudd commented on source/winAPI/constants.py:
```diff
@@ -3,9 +3,14 @@
 
 
+#: Verification of the architecture of the running system
+is_64Bit = winVersion.getWinVer().processorArchitecture.endswith("64")
```
We use capitals when naming constants

```suggestion
IS_64_BIT: bool = winVersion.getWinVer().processorArchitecture.endswith("64")
```

seanbudd commented on source/JABHandler.py:
```diff
@@ -6,6 +6,7 @@
 
 from enum import IntEnum, IntFlag
 import os
+from winAPI.constants import is_64Bit
```
```suggestion
from winAPI.constants import IS_64_BIT
```

seanbudd commented on source/JABHandler.py:
```diff
@@ -81,7 +82,8 @@ def _fixBridgeFunc(restype,name,*argtypes,**kwargs):
 
-class JOBJECT64(c_int64):
+# If the machine is 64-bit, use c_int64, otherwise use c_int as a parameter.
+class JOBJECT64(c_int64 if is_64Bit else c_int):
```
```suggestion
class JOBJECT64(c_int64 if IS_64_BIT else c_int):
```

seanbudd commented on source/JABHandler.py:
```diff
@@ -836,10 +840,12 @@ def enableBridge():
 
 def initialize():
-	global bridgeDll, isRunning
+	global bridgeDll, is_64Bit, isRunning
```
This is not necessary

```suggestion
	global bridgeDll, isRunning
```

seanbudd commented on source/JABHandler.py:
```diff
@@ -836,10 +840,12 @@ def enableBridge():
+	global bridgeDll, is_64Bit, isRunning
+	# If the system is 64-bit, load the dll that we have in the NVDA distribution.
+	# Otherwise, it loads the one on the 32-bit system, which does not have the -32 suffix.
+	correctDll = "windowsaccessbridge-32.dll" if is_64Bit else "windowsaccessbridge.dll"
```
```suggestion
	correctDll = "windowsaccessbridge-32.dll" if IS_64_BIT else "windowsaccessbridge.dll"
```

seanbudd commented on user_docs/en/changes.md:
```diff
@@ -41,6 +44,7 @@ It is especially useful to read the error location markers in tracebacks. (#1632
 * When a `gainFocus` event is queued with an object that has a valid `focusRedirect` property, the object pointed to by the `focusRedirect` property is now held by `eventHandler.lastQueuedFocusObject`, rather than the originally queued object. (#15843)
 * NVDA will log its executable architecture (x86) at startup. (#16432, @josephsl)
 * `wx.CallAfter`, which is wrapped in `monkeyPatches/wxMonkeyPatches.py`, now includes proper `functools.wraps` indication. (#16520, @XLTechie)
+* Added a variable called `is_64Bit` in `winAPI.constants` to determine the architecture of the running system. (#16330, #16557)
```
This is not neccessary

```suggestion
```

seanbudd commented on sconstruct:
```diff
@@ -217,7 +217,11 @@ envArm64['projectResFile'] = resFile
 # Copy in some other dependencies.
-jabDll = "windowsaccessbridge-32.dll"
+jabDll32 = "windowsaccessbridge-32.dll"
+Command(sourceLibDir.File(jabDll32),
+	env.Dir("#include/javaAccessBridge32").File(jabDll32),
+	Copy("$TARGET", "$SOURCE"))
```
```suggestion
Command(
	sourceLibDir.File(jabDll32),
	env.Dir("#include/javaAccessBridge32").File(jabDll32),
	Copy("$TARGET", "$SOURCE")
)
```

seanbudd reviewed (approved):
Thanks, I'm going to apply some minor changes. When the final build completes, please do a final round of testing and then we can merge

seanbudd commented on source/winAPI/constants.py:
```diff
@@ -3,9 +3,14 @@
 import enum
 
 
+#: Verification of the architecture of the running system
```
```suggestion
```

seanbudd committed:
Apply suggestions from code review

seanbudd committed:
Apply suggestions from code review

seanbudd committed:
Update source/JABHandler.py

seanbudd commented:
@FelipeZanabria - can you test this build out: https://ci.appveyor.com/api/buildjobs/lxn6iv0md5en4cd8/artifacts/output%2Fnvda_snapshot_pr16557-32144%2C803cdcf3.exe

seanbudd commented:
Some wider community testing would be also be welcome, for using Java Access Bridge Applications on various combinations of:
- 32/64bit system architecture
- 32/64bit software
- Windows 8.1, 10, 11

FelipeZanabria committed:
Merge branch 'master' into javaAccessBridgeFix

FelipeZanabria commented:
> @FelipeZanabria - can you test this build out: https://ci.appveyor.com/api/buildjobs/lxn6iv0md5en4cd8/artifacts/output%2Fnvda_snapshot_pr16557-32144%2C803cdcf3.exe

With my W7 it is impossible to test this snapshot, but 3 days ago when you updated the submodule, I checked if the two Java dlls were in the installer. They weren't there, so I looked in sconscript, and after seeing that the first dll was there, I added the last one, waited for the compilation and it was included there.
Also before publishing each commit I tested the changes in my setup.

josephsl commented:
Hello Felipe,

Do you have access to a computer running Windows 8.1 or later, preferably Windows 10 22H2 (build 19045) or later? That way you can do a live test (running the modified NVDA executable) of the modifications from this pull request instead of doing compilation and library inclusion tests.

As I stated earlier, testing NVDA pull requests on Windows 7 is really questionable because NVDA 2024.1 and later (including the master branch code) does NOT support Windows 7 in any form or shape. This is an important thing to remember because the best way to see if your PR is working is doing live tests - running the modified NVDA with Java apps (with JAB applied) yourself. I think testing the inclusion of the library is a useful strategy on Windows 7, but by not considering live tests, the effectiveness and impact of the pull request is diminished (I'm sorry to say). So as I also stated, before writing another pull request, PLEASE get a system running updated Windows releases such as 10 and 11 so you can do live tests, improving the effectiveness and impact of pull requests.

Thanks.

FelipeZanabria committed:
Update changes.md to resolve conflicts

FelipeZanabria commented:
@josephsl I don't have it, but I'm going to write to the mailing list in Spanish inviting them to try the changes.

josephsl commented:
Hi Felipe,

I see. Thanks for the pull request, and please do use Windows 10 or 11 yourself while writing the next pull request.

Thakns.

britechguy commented:
I honestly cannot believe we are talking about making changes for Windows 7, which is out of support and has been for some time now, and when NVDA itself no longer supports anything but Windows 10 and 11, with Windows 10 having a sunset date just a hair over a year from now.

It is long past time to stop even considering Windows 7, and to a lesser extent 32-bit, and move on.  You don't keep software vital and up to date by tortured attempts at backward compatibility, and particularly with out of support versions of Windows.

Adriani90 commented:
@mwhapples have you any chance to test this PR on some 32bit java applications on 32 / 64 bit systems?

At least on my Windows 11 on a 64 bit system it works well with the 64bit java configuration window which indicates that the java access bridge for 64 bit works aas expected, so this PR doesn't seem to have any effect on 64 bit.
So if we can test that for 32bit works as well, then I guess it should be fine.

Adriani90 commented:
I also tested with JRE 8.411 32bit, it still works fine on my 64 bit system.

So I think we need here testing on pure 32 bit systems I guess.

@FelipeZanabria do you have a test application we can test with? Or can you give some recommendations?

FelipeZanabria commented:
> I also tested with JRE 8.411 32bit, it still works fine on my 64 bit system.
> 
> So I think we need here testing on pure 32 bit systems I guess.
> 
> @FelipeZanabria do you have a test application we can test with? Or can you give some recommendations?

@Adriani90 Thanks for trying and commenting.
The Java control panel itself is enough, although I also tried the Linux sampler for Windows 32-bit.
https://download.linuxsampler.org/packages/win32/snapshots/linuxsampler_20240326_setup.exe
The installer covers both architectures, so you must have a 32-bit system for x32 to be installed. In the components to be installed, you must check Jsampler, which is the Java frontend application.

seanbudd commented:
@britechguy - these changes are for 32bit systems, which are still in support for now. Windows 7 is not under support any more

XLTechie commented:
@britechguy To be clear, this is not a change "for Windows 7". The W7 conversation was in relation to the contributor's build environment, not the contribution itself.

Additionally, NVDA still supports Windows 8.1 (and above), not only 10 and 11.

seanbudd commented:
>I don't have it, but I'm going to write to the mailing list in Spanish inviting them to try the changes.

@FelipeZanabria - can you link to this thread?
When we get testing confirmed with a 32bit system I think we can merge

FelipeZanabria commented:
Here is the thread:
https://groups.google.com/g/nvda-es/c/2JwIUPg_Lo0

FelipeZanabria commented:
@seanbudd it seems that no one on the Spanish mailing list uses 32-bit Windows, except for one who also uses w7, which doesn't help moving forward.
Perhaps you could ask on the international mailing list.
What do you think? Could someone do it? Thank you.

seanbudd commented:
@FelipeZanabria - could you send a message to the mailing list?

FelipeZanabria commented:
I didn't want to subscribe just for that, but I'm going to find it and do this.

FelipeZanabria commented:
I didn't get an ansuer on the international list either. Even the moderators put a note warning against replying in the group, since almost no one has 32-bit hardware. The recommendation was that they write to me privately. No one has done this either.
What do you think about it? Should this be closed without implementing the changes? Should another PR be opened just to update constants.py with IS_64BIT?

seanbudd commented:
I think unfortunately we will just leave this as blocked until testing can be done, and close eventually if none can be done

seanbudd commented:
@ivnc you reported on #10842 that you could reproduce the issue on Windows 10. Can you test that this fix works for you?

seanbudd commented:
@FelipeZanabria feel free to open a separate PR to handle https://github.com/nvaccess/nvda/issues/16330

wmhn1872265132 commented:
Tested according to the Testing strategy on a 32-bit virtual machine, and it is in line with expectations.

FelipeZanabria commented:
> Tested according to the Testing strategy on a 32-bit virtual machine, and it is in line with expectations.

Maybe it is not necessary to create another PR.

ivnc commented:
> @ivnc you reported on #10842 that you could reproduce the issue on Windows 10. Can you test that this fix works for you?

Yes, we had an user who reported this in Windows 10. I've just located that ticket, let me contact him and will report back as soon as he replies.
Thanks.

seanbudd commented:
@ivnc - any updates on testing [the build](https://ci.appveyor.com/api/buildjobs/lxn6iv0md5en4cd8/artifacts/output%2Fnvda_snapshot_pr16557-32144%2C803cdcf3.exe) out?

Thanks for reaching out

ivnc commented:
No, so far there has been no response, I've just contacted the user again. Hopefully there'll be one soon, I keep you updated.

seanbudd commented:
@FelipeZanabria - could you resolve merge conflicts?

seanbudd commented:
Closing as abandoned
